hw4/02-sessions.js

```diff
@@ -6,12 +6,49 @@ const port = process.env.PORT || 5001;
 // Add your code here
 
 // Use the express-session module
-app.use(/** ... */);
+app.use(
+	session({
+		secret: '123456789',
+		saveUninitialized: true,
+		cookie: { maxAge: 86400 },
+		resave: false,
+	})
+);
+app.use((req, res, next) => {
+	res.status(200);
+	res.header({ 'Content-Type': 'text/html' });
+	res.write(`<p>Currently on route:${req.url}</p>`);
+	if (!req.session.routes) {
+		req.session.routes = [`<ul><p>Previously visited:</p></ul>`];
+	} else {
+		req.session.routes.forEach((route) => res.write(route));
+		req.session.routes.push(`<li>${req.url}</li>`);
+	}
+	next();
+});
 
 app.get('/', (req, res) => {
-  // Add your code here
+	// Add our code here
+	res.write(`<p>Welcome to http://${req.headers.host}</p>`);
+	res.end();
+});
+
+app.get('/main', (req, res) => {
```

Comment:
Your code looks good. Can I ask what these extra routes with `res.end()` accomplish? I think the solution should work on any route, so I'm not sure what these are for, thanks!